optimize/backend/src/test/java/io/camunda/optimize/service/ProcessOverviewServiceTest.java

```diff
@@ -0,0 +1,194 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.optimize.dto.optimize.IdentityDto;
+import io.camunda.optimize.dto.optimize.IdentityType;
+import io.camunda.optimize.dto.optimize.UserDto;
+import io.camunda.optimize.dto.optimize.query.definition.DefinitionWithTenantIdsDto;
+import io.camunda.optimize.dto.optimize.query.processoverview.ProcessDigestDto;
+import io.camunda.optimize.dto.optimize.query.processoverview.ProcessDigestRequestDto;
+import io.camunda.optimize.dto.optimize.query.processoverview.ProcessOverviewDto;
+import io.camunda.optimize.dto.optimize.query.processoverview.ProcessOverviewResponseDto;
+import io.camunda.optimize.dto.optimize.query.processoverview.ProcessUpdateDto;
+import io.camunda.optimize.service.db.reader.ProcessOverviewReader;
+import io.camunda.optimize.service.db.writer.ProcessOverviewWriter;
+import io.camunda.optimize.service.digest.DigestService;
+import io.camunda.optimize.service.identity.AbstractIdentityService;
+import io.camunda.optimize.service.security.util.definition.DataSourceDefinitionAuthorizationService;
+import jakarta.ws.rs.ForbiddenException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+public class ProcessOverviewServiceTest {
+
+  private final DefinitionService definitionService = mock(DefinitionService.class);
+  private final DataSourceDefinitionAuthorizationService definitionAuthorizationService =
+      mock(DataSourceDefinitionAuthorizationService.class);
+  private final ProcessOverviewWriter processOverviewWriter = mock(ProcessOverviewWriter.class);
+  private final ProcessOverviewReader processOverviewReader = mock(ProcessOverviewReader.class);
+  private final AbstractIdentityService identityService = mock(AbstractIdentityService.class);
+  private final KpiService kpiService = mock(KpiService.class);
+  private final DigestService digestService = mock(DigestService.class);
+
+  private final ProcessOverviewService processOverviewService =
+      new ProcessOverviewService(
+          definitionService,
+          definitionAuthorizationService,
+          processOverviewWriter,
+          processOverviewReader,
+          identityService,
+          kpiService,
+          digestService);
+
+  @Test
+  void shouldReturnProcessOverviewsWhenUserIsAuthorized() {
+    // given
+    final String userId = "user123";
+    final String locale = "en";
+    final String processKey = "processKey1";
+    final String processName = "Process 1";
+
+    final List<String> tenants = new ArrayList<>();
+    tenants.add("1");
+
+    final DefinitionWithTenantIdsDto definitionDto =
+        new DefinitionWithTenantIdsDto(processKey, processName, PROCESS, tenants, Set.of());
+    when(definitionService.getAllDefinitionsWithTenants(PROCESS))
+        .thenReturn(List.of(definitionDto));
```

Comment:
You could also do a test where this method returns no definitions, to ensure that the overview list is also empty

Reply:
fixed